### PR TITLE
HammerLeapp commands

### PR DESCRIPTION
**Pull request by stejskalleos**
Requires latest Leapp master branch - https://github.com/oamg/foreman_leapp

### CODE REVIEW

File: lib/hammer_cli_foreman_leapp.rb [hunk 1]
```diff
@@ -7,4 +7,10 @@
 require 'hammer_cli_foreman_leapp/preupgrade_reports'
 
 module HammerCLIForemanLeapp
+  HammerCLI::MainCommand.lazy_subcommand(
+    'preupgrade-reports',
```
Comment (adamruzicka) [inline lib/hammer_cli_foreman_leapp.rb]: hammer usually uses nouns in singular for subcommand names

File: .gitignore [hunk 2]
```diff
@@ -0,0 +1,6 @@
+# Ruby
+Gemfile.lock
+Gemfile.local
+Gemfile.local.rb
+pkg
+.bundle
```
Comment (adamruzicka) [inline .gitignore]: Missing ending newline

**Review by adamruzicka — COMMENTED** [review]
Two nitpicks, overall looks good to me.

File: lib/hammer_cli_foreman_leapp/preupgrade_reports.rb [hunk 3]
```diff
@@ -2,5 +2,50 @@
...
+    class JobInvocationCommand < HammerCLIForeman::ListCommand
+      action :job_invocation
+      command_name 'job-invocation'
+
+      output do
+        field :id, _('Id')
+        field :reported_at, _('Reported at'), Fields::Date
+        field :job_invocation_id, _('Job Invocation Id')
+        field :host_id, _('Host Id')
+      end
```
Comment (adamruzicka) [inline lib/hammer_cli_foreman_leapp/preupgrade_reports.rb]: This is exactly same as the block on line 8, could we somehow reuse the definition?

File: lib/hammer_cli_foreman_leapp/preupgrade_reports.rb [hunk 4]
```diff
@@ -2,5 +2,50 @@
...
+      output PreupgradeReportsCommand::ListCommand.output_definition do
+        collection :preupgrade_report_entries, _('Entries'), hide_blank: true do
+          field :id, _('Id')
+          field :hostname, _('Host name')
+          field :title, _('Title')
+          field :actor, _('Actor')
+          field :audience, _('Audience')
+          field :severity, _('Severity')
+          field :tags, _('Tags'), Fields::List
+          field :summary, _('Summary'), Fields::LongText
```
Comment (ofedoren) [inline lib/hammer_cli_foreman_leapp/preupgrade_reports.rb]: I'd rather use `Fields::Text`. `Fields::LongText` is more for backward compatibility.

File: lib/hammer_cli_foreman_leapp/preupgrade_reports.rb [hunk 5]
```diff
@@ -2,5 +2,43 @@
...
+        end
+      end
+
+      build_options
+    end
+
+    class JobInvocationCommand < HammerCLIForeman::ListCommand
```
Comment (ofedoren) [inline lib/hammer_cli_foreman_leapp/preupgrade_reports.rb:35]: I'd rather add this command as a subcommand for `HammerCLIForemanRemoteExecution::JobInvocation` so it's better structured, but I'm ok with this. Let's not complicate things with that for now.

**Review by ofedoren — CHANGES_REQUESTED** [review]
Just one nitpick for output fields and ACK from me :)

**Review by ofedoren — APPROVED** [review]
ACK from me. I don't see any reason why it shouldn't be merged :) Good job, @stejskalleos!

**Comment by adamruzicka** [comment]
Thank you both!

**Pull request merged**